src/block/block_ckpt.c

```diff
@@ -1084,6 +1084,52 @@ __wt_block_checkpoint_resolve(WT_SESSION_IMPL *session, WT_BLOCK *block, bool fa
     return (ret);
 }
 
+/*
+ * __wti_block_checkpoint_extlist_dump --
+ *     Dump all of the available checkpoints extent lists, excluding the specified offset. Extent
+ *     offsets should never be 0, as that is the offset of the file header. Therefore, an
+ *     exclude_offset value of 0 can be set to ensure that all extent blocks are dumped.
+ */
+int
+__wti_block_checkpoint_extlist_dump(
+  WT_SESSION_IMPL *session, WT_BLOCK *block, wt_off_t exclude_offset)
```

Comment:
What should happen if there are more than one corrupt blocks?

Reply:
Yes but dumping extlists involves reading blocks where we can encounter another bad block. Can we end up in a recursion with 2 bad blocks?

Reply:
Very good point, if there are two bad extent blocks we'd recursively alternate trying to read each.